exonum/src/node/mod.rs

```diff
@@ -934,10 +978,26 @@ impl Node {
         };
         self.run_handler(&handshake_params)?;
 
-        // Stop all api handlers.
-        for mut handler in api_handlers {
-            handler.close().unwrap();
+        debug!("Handler stopped");
+
+        // Stop all APIs
+        for api_sys_addr in api_sys_addrs {
+            api_sys_addr?
+                .send(actix_web::server::StopServer { graceful: true })
+                .wait()?
+                .map_err(|_| format_err!("Unable to send StopServer message"))?;
         }
+        actix_sys_addr.send(actix::msgs::SystemExit(0)).wait()?;
+
+        debug!("Sent stop signal to the actix system");
+        actix_api_thread.join().map_err(|e| {
+            format_err!(
+                "Unable to join actix web api thread, an error occurred: {:?}",
+                e
+            )
+        })??;
```

Comment:
I'm not even sure that it's legit.

So general impression after this method - it's too overloaded and must be splitted in several functions.
Please avoid `addr` naming, `address` isn't much longer but increases readability in cases like `for api_sys_addr in api_sys_addrs`